packets.go

```diff
@@ -224,36 +219,34 @@ func (mc *mysqlConn) readInitPacket() ([]byte, string, error) {
 		//
 		// The official Python library uses the fixed length 12
 		// which seems to work but technically could have a hidden bug.
-		cipher = append(cipher, data[pos:pos+12]...)
+		authData = append(authData, data[pos:pos+12]...)
 		pos += 13
 
 		// EOF if version (>= 5.5.7 and < 5.5.10) or (>= 5.6.0 and < 5.6.2)
 		// \NUL otherwise
 		if end := bytes.IndexByte(data[pos:], 0x00); end != -1 {
-			pluginName = string(data[pos : pos+end])
+			plugin = string(data[pos : pos+end])
 		} else {
-			pluginName = string(data[pos:])
+			plugin = string(data[pos:])
 		}
 
 		// make a memory safe copy of the cipher slice
 		var b [20]byte
-		copy(b[:], cipher)
-		return b[:], pluginName, nil
+		copy(b[:], authData)
+		return b[:], plugin, nil
```

Comment:
`b` instead of `b[:]`.  (While old code was same...)

Reply:
`copy` requires slices as parameters but `b` is an array and not a slice.